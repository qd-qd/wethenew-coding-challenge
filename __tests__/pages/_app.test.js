import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import sneaker from '@test/fixtures/sneaker.json';
import MyApp from '@pages/_app';
import Home from '@pages';

const sneakerList = new Array(3)
  .fill(sneaker)
  .map((value, index) => ({ ...value, id: `${index}` }));

describe('App', () => {
  test('displays greeting', () => {
    render(<MyApp Component={Home} pageProps={{ sneakers: sneakerList }} />);

    expect(screen.getByTestId('header')).toBeDefined();
    expect(screen.getByTestId('card-list')).toBeDefined();
  });

  test("doesn't break snapshot", () => {
    const { container } = render(
      <MyApp Component={Home} pageProps={{ sneakers: sneakerList }} />,
    );

    expect(container).toMatchSnapshot();
  });
});
