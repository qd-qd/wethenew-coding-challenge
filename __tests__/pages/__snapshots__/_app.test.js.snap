// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App doesn't break snapshot 1`] = `
<div>
  <header
    class="header"
    data-testid="header"
  >
    <div
      class="header-content"
    >
      <img
        alt="logo"
        class="logo"
        src="/logo.png"
      />
    </div>
  </header>
  <main
    class="main"
  >
    <h1
      class="title"
    >
      <span
        class="title__count"
      >
        3 
      </span>
      sneakers
    </h1>
    <hr
      class="separator"
    />
    <section
      class="container"
      data-testid="card-list"
    >
      <a
        data-testid="card-link"
        href="/fake-url"
      >
        <div
          class="container"
        >
          <img
            alt="The Fake shoes sneaker by Jest dude"
            class="image"
            src="/sneakers-placeholder.png"
          />
          <div
            class="section-infos"
          >
            <p
              class="name"
            >
              Fake shoes
            </p>
            <p
              class="price"
            >
              999
              <span>
                €
              </span>
            </p>
          </div>
          <p
            class="brand"
          >
            Jest dude
          </p>
        </div>
      </a>
      <a
        data-testid="card-link"
        href="/fake-url"
      >
        <div
          class="container"
        >
          <img
            alt="The Fake shoes sneaker by Jest dude"
            class="image"
            src="/sneakers-placeholder.png"
          />
          <div
            class="section-infos"
          >
            <p
              class="name"
            >
              Fake shoes
            </p>
            <p
              class="price"
            >
              999
              <span>
                €
              </span>
            </p>
          </div>
          <p
            class="brand"
          >
            Jest dude
          </p>
        </div>
      </a>
      <a
        data-testid="card-link"
        href="/fake-url"
      >
        <div
          class="container"
        >
          <img
            alt="The Fake shoes sneaker by Jest dude"
            class="image"
            src="/sneakers-placeholder.png"
          />
          <div
            class="section-infos"
          >
            <p
              class="name"
            >
              Fake shoes
            </p>
            <p
              class="price"
            >
              999
              <span>
                €
              </span>
            </p>
          </div>
          <p
            class="brand"
          >
            Jest dude
          </p>
        </div>
      </a>
    </section>
  </main>
</div>
`;
